### PR TITLE
Improve error message when max-iters is set to 1

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

File: src/Translation_unit.ml [hunk 1]
```diff
@@ -236,6 +238,18 @@ let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
       | Warning50 l ->
           List.iter l ~f:(fun (l, w) -> !Location.warning_printer l fmt w)
       | exn -> Format.eprintf "%s\n%!" (Exn.to_string exn) )
+  | Not_stable i when i <= 1 ->
+      Format.eprintf "%s: %S needs to be formatted. ([max-iters = 1])\n%!"
```
Comment (jberdine) [inline src/Translation_unit.ml]: maybe reword: s/needs to be/was not already/

File: src/Translation_unit.ml [hunk 2]
```diff
@@ -102,7 +102,11 @@ let dump xunit dir base suf ext ast =
     xunit.printast (Caml.Format.formatter_of_out_channel oc) ast ;
     Out_channel.close oc )
 
-type result = Ok | Invalid_source of exn | Ocamlformat_bug of exn
+type result =
+  | Ok
+  | Invalid_source of exn
+  | Not_stable of int
```
Comment (jberdine) [inline src/Translation_unit.ml]: maybe `Unstable`?

**Comment by hhugo** [comment]
I've added explicit `--max-iters` whenever required. The default value it set by environment variable. To use a `.ocamlformat` file, we would first need to address https://github.com/ocaml-ppx/ocamlformat/issues/309.
